### PR TITLE
Add more hooks to support structlog contextvars

### PR DESCRIPTION
**Pull request by kafonek**
`context_hook` that's called by inbound worker, outbound worker, and poll worker during their respective `while True` loops. Intended to be used to set `structlog.contextvars.bind_contextvars` in application code. Can be defined as class methods or attached to an instance.

Related PA issue https://github.com/noteable-io/planar-ally/pull/243

### CODE REVIEW

**Comment by kafonek** [comment]
Current dependencies on/for this PR:
* main
  * **PR #36** <a href="https://app.graphite.dev/github/pr/noteable-io/sending/36" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/noteable-io/sending/36?utm_source=stack-comment).

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - CI/CD files and noxfile syntax
 - `WebsocketManager` Backend
 - New extra install `-E websockets`, additionally a convenience `-E all` option
-- `WebsocketManager` saves response headers on connect
```
Comment (tizz98) [inline CHANGELOG.md:15]: did you mean to remove this line?

**Review by tizz98 — APPROVED** [review]
Generally looks good. I wonder if a future refactor could use something like event emitter pattern rather than defining hook functions (https://pypi.org/project/PyEventEmitter/)

File: CHANGELOG.md (hunk 1 above)
Comment (kafonek) [reply CHANGELOG.md:15]: Yeah, I removed the response-header-saving functionality in favor of `connect_hook`s on the websocket manager

**Pull request merged**